### ISSUE TITLE
C error: related to file opening based on condition

### ISSUE THREAD

**Issue by VinWare** [vlang/v#11474]
**V doctor:**
```
OS: linux, "Garuda Linux"
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-7020U CPU @ 2.30GHz
CC version: cc (GCC) 11.1.0

getwd: /home/vineet/Developer/V/coreutils/src/base32
vmodules: /home/vineet/.vmodules
vroot: /home/vineet/Developer/V/v
vexe: /home/vineet/Developer/V/v/v
vexe mtime: 2021-09-11 13:16:52
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 72089c4.ccf6285

Git version: git version 2.33.0
Git vroot status: weekly.2021.36-45-gccf6285f
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg temp.v`
```v
import os

file := ['a']
f := if file.len == 0 || file[0] == '-' { os.stdin() } else { os.open(file[0]) or { exit(1) } }

```

**What did you expect to see?**

No error, or at most a V compilation error

**What did you see instead?**
```
temp.v:4:1: warning: unused variable: `f`
    2 | 
    3 | file := ['a']
    4 | f := if file.len == 0 || file[0] == '-' { os.stdin() } else { os.open(file[0]) or { exit(1) } }
      | ^
==================
/tmp/v/temp.2548474834843537235.tmp.c:14195: error: ')' expected (got "_t1")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

**Comment by VinWare** [comment]
The following code compiles, which should be equivalent, except for the additional function call:
```
import os

fn get_file(file []string) os.File {
	if file.len == 0 || file[0] == '-' {
		return os.stdin()
	} else {
		return os.open(file[0]) or { exit(1) }
	}
}

fn main() {
	file := ['a']
	f := get_file(file)
}
```

**Comment by spytheman** [comment]
Fixed in cc4af23 .
